enterprise/backend/test/metabase_enterprise/audit_app/api/collection_test.clj

```diff
@@ -29,11 +36,67 @@
         (testing "Instance Analytics Collection should not show up when audit-app isn't enabled."
           (testing "GET /api/collection"
             (is (nil?
-                   (->> (mt/user-http-request :rasta :get 200 "collection")
+                   (->> (mt/user-http-request :crowberto :get 200 "collection")
                         last
                         :type))))
           (testing "GET /api/collection/test"
             (is (= nil
-                   (->> (mt/user-http-request :rasta :get 200 "collection/tree")
+                   (->> (mt/user-http-request :crowberto :get 200 "collection/tree")
                         last
                         :type)))))))))
+
+(defn instance-analytics-collection-names
+  "Gather instance-analytic type collections and their children (who may or may-not have type=instance-analytics)."
+  []
+  (if-not config/ee-available?
+    #{}
+    (let [colls (mapv #(select-keys % [:id :name :location :type]) (t2/select Collection :archived false))
+          id->coll (m/index-by :id colls)
+          collection-tree (collection/collections->tree {} colls)]
+      (->> (loop [[tree & coll-tree] collection-tree
+                  ia-ids #{}]
+             (cond (not tree) ia-ids
+                   (= "instance-analytics" (:type tree)) (let [ids (transient #{})]
+                                                           (walk/postwalk
+                                                            (fn [x] (when (and (map? x) (:id x)) (conj! ids (:id x))) x)
+                                                            tree)
+                                                           (recur coll-tree (into ia-ids (persistent! ids))))
+                   ;; TODO: put my children onto the end of the coll-tree, then recur like normal
+                   :else (recur (concat coll-tree (:children tree)) ia-ids)))
+           (mapv (comp :name id->coll))))))
+
+(deftest list-collections-visible-collections-test
```

Comment:
Moved this test to enterprise namespace, and also added `with-audit-db-restoration`.